### PR TITLE
Fix a bug in username minimum length transformation and set minimum length to 1

### PR DESCRIPTION
**Pull request by bernt-matthias**
1. Fix the `ljust` call in the transformation

2. Set minimum length to 1:

Two would be sufficient to make at least one `Yu`, `Li`, ... per Galaxy instance happy :) Korea seem to allow for single letter names.

Anyway, I just want to make Mr T. happy :)

The serious background is that this seems to be an easy way to solve the following problem: If a user is registered some transformation of the username happens, e.g. it is filled by - to enforce the minlength requirement. On LDAP managed systems that submit jobs as real users this leads to the problem that users with to short names can't submit jobs.

May replace https://github.com/galaxyproject/galaxy/pull/13177

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/security/validate_user_input.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def transform_publicname(publicname):
     if publicname not in ['None', None, '']:
         publicname = publicname.lower()
         publicname = re.sub(VALID_PUBLICNAME_SUB, FILL_CHAR, publicname)
```
Comment (nsoranzo) [inline lib/galaxy/security/validate_user_input.py]:
```suggestion
    if not publicname:
        raise ValueError("Public name cannot be empty")
    publicname = publicname.lower()
    publicname = re.sub(VALID_PUBLICNAME_SUB, FILL_CHAR, publicname)
```

File: lib/galaxy/security/validate_user_input.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def transform_publicname(publicname):
...
-        publicname = publicname.ljust(PUBLICNAME_MIN_LEN + 1, FILL_CHAR)[:PUBLICNAME_MAX_LEN]
+        publicname = publicname.ljust(PUBLICNAME_MIN_LEN, FILL_CHAR)[:PUBLICNAME_MAX_LEN]
```
Comment (nsoranzo) [inline lib/galaxy/security/validate_user_input.py]:
```suggestion
    publicname = publicname[:PUBLICNAME_MAX_LEN]
```

File: lib/galaxy/security/validate_user_input.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 EMAIL_MAX_LEN = 255
 
 # Public name validity parameters
-PUBLICNAME_MIN_LEN = 3
+PUBLICNAME_MIN_LEN = 1
```
Comment (nsoranzo) [inline lib/galaxy/security/validate_user_input.py]:
```suggestion
```

and replace the test in `validate_publicname_str()` with:
```python
    if not publicname:
        raise ValueError("Public name cannot be empty")
```

File: test/unit/app/managers/test_UserManager.py [hunk 4]
```diff
@@ -289,7 +289,7 @@ def test_username_validation(self):
 
         self.log("usernames must be long enough and with no non-hyphen punctuation")
         exception = self._assertRaises_and_return_raised(base_manager.ModelDeserializingError,
-            self.deserializer.deserialize, user, {'username': 'ed'}, trans=self.trans)
+            self.deserializer.deserialize, user, {'username': ''}, trans=self.trans)
         self.assertTrue('Public name must be at least' in str(exception))
```
Comment (nsoranzo) [inline test/unit/app/managers/test_UserManager.py]:
```suggestion
        self.assertTrue('Public name cannot be empty' in str(exception))
```

**Comment by bernt-matthias** [comment]
@nsoranzo Applied your suggestions. Lets see if we get problems with `"None"` that we do not check for now. 

But this might make Mr/Mrs None happy, LOL.

**Comment by hexylena** [comment]
> None

Do we have test cases for user names we can write? Or, can we use these new potential values in some of the existing tests that exercise the UI?

**Comment by jmchilton** [comment]
The public name doesn't have to be a person's actual name right? I am not sure the short public names are a good idea. I am fine allowing it for LDAP backed things - maybe that should be behind a configuration option though? 

I'm not sure it is a huge issue to me either way though. There is an API test for the name being too short that is failing though.

**Comment by mvdbeek** [comment]
As a bug it can go in whenever we're ready, but moving to 22.05 for now

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
I've removed the `status/WIP` label in accordance with @bernt-matthias's comment on the committers channel that this is ready for review/merge.

**Pull request merged**